code/protw3.c

```diff
@@ -16,6 +16,17 @@
 SRCID(protw3, "$Id$");
 
 
+/* Save and restore the value in GetLastError on Windows. See
+ * <design/thread-manager#.sol.thread.lasterror>
+ *
+ * These macros must be used after #include "mpswin.h" so that
+ * GetLastError() and SetLastError() are defined.
+ */
+
+#define LAST_ERROR_SAVE BEGIN DWORD _saved_last_error = GetLastError(); BEGIN
```

Comment:
Why use macros?

Reply:
I used the ERRNO_SAVE and ERRNO_RESTORE as a template here initially. But since it is only used in one location at the moment I agree that the macro-free version is better. I will fix that.